backend/de.metas.async/src/main/java/de/metas/lock/api/impl/AbstractLockDatabase.java

```diff
@@ -326,6 +328,69 @@ public final String getLockedWhereClause(final Class<?> modelClass, final String
 				.filter(TypedSqlQueryFilter.of(lockedRecordsSQL));
 	}
 
+	@Override
+	@NonNull
+	public final <T> List<T> retrieveAndLockMultipleRecords(@NonNull final IQuery<T> query,@NonNull final Class<T> clazz)
+	{
+		final int maxLockRetries = 50;
+		final ILockCommand lockCommand = new LockCommand(this)
+				.setOwner(LockOwner.NONE);
+
+		final ArrayList<T> lockedModels = new ArrayList<>();
+
+		int retryCounter = 0;
+		while (retryCounter < maxLockRetries)
```

Comment:
I don't see a point in checking this more than once. i mean, what are the odds?
(note: there is not even a delay or something in between... NOT that I suggest introducing one, no way!)

more, I think the method shall be called something like `lockIfNotAlreadyLocked` because to me that's what it does.

Reply:
PS: i saw that on retrieveAndLock we do the same.... but still, i don't see the point. @metas-ts  do u?

Reply:
i also can't see why. the other retry ist there sinse at least 2015..